### PR TITLE
add bad cookie pressure point

### PR DESCRIPTION
**Pull request by shubham808**

### CODE REVIEW

File: nfs/src/rpc_task.cpp [hunk 1]
```diff
@@ -3368,7 +3397,7 @@ static void readdirplus_callback(
                      * didn't return eof, this means the directory shrank in the
                      * server. To be safe, invalidate the cache.
                      */
-                    AZLogWarn("[{}] readdirplus_callback{}: Directory shrank in "
+                    AZLogWarn("[{}] readdirplus_callback {}: Directory shrank in "
```
Comment (linuxsmiths) [inline nfs/src/rpc_task.cpp:3391]: why did u add a space before {}

File: nfs/src/rpc_task.cpp [hunk 2]
```diff
@@ -3132,6 +3146,21 @@ static void readdirplus_callback(
...
+             */
+            if (inject_error() && status == 0) {
+                AZLogWarn("[{}] PP: bad cookie at offset {} ",
+                          is_reenumerating ? "(R)" : "",
+                          dir_ino, task->rpc_api->readdir_task.get_offset());
+
+                (res)->status = NFS3ERR_BAD_COOKIE;
+                status = task->status(rpc_status, NFS_STATUS(res));
+            }
+#endif
```
Comment (linuxsmiths) [inline nfs/src/rpc_task.cpp]: let's make this a macro like INJECT_JUKEBOX() and use it after INJECT_JUKEBOX(), to not muddle the code

File: nfs/src/rpc_task.cpp [hunk 3]
```diff
@@ -2831,6 +2831,21 @@ static void readdir_callback(
     const bool is_reenumerating =
         (task->rpc_api->readdir_task.get_target_offset() != 0);
 
+#ifdef ENABLE_PRESSURE_POINTS
+            /*
+             * Bad cookie pressure point. Random bad cookie errors which
+             * can cause multiple re-enumerations.
+             */
+            if (inject_error() && status == 0) {
```
Comment (linuxsmiths) [inline nfs/src/rpc_task.cpp]: add offset!=0 to this and reinstate the assert below

File: nfs/src/rpc_task.cpp (hunk 1 above)
Comment (shubham808) [reply nfs/src/rpc_task.cpp:3391]: looked weird in the logs like this

**Review by linuxsmiths — APPROVED** [review]
merging, will audit and review post merge

**Pull request merged**